### ISSUE TITLE
merge endDate & endDateTime; startDate & startDateTime

### ISSUE THREAD

**Issue by VladimirAlexiev** [schemaorg/schemaorg#1512]
Consider the following dichotomy :
``` turtle
schema:endDate a rdf:Property ;
    schema:domainIncludes schema:CreativeWorkSeason, schema:CreativeWorkSeries,
        schema:DatedMoneySpecification, schema:Event, schema:Role ;
    schema:rangeIncludes schema:Date, schema:DateTime .
schema:endTime a rdf:Property ;
    schema:domainIncludes schema:Action, schema:FoodEstablishmentReservation ;
    schema:rangeIncludes schema:DateTime ;
```
IMHO a property should reflect semantics, not granularity of representation.
- How do you know the granularity expressed in the prop name is necessarily the correct one?
- According to the prop name: why Event is "measured" in days but Action in seconds?
- in fact `endDate rangeIncludes DateTime` shows that the name `endDate` is incorrect
- when applying these props to future classes, one would have to decide whether to use one or the other, which is IMO an unnecessary complication.

The same applies to `startDate[Time]`.

Other props that commit to a granularity include the following. I marked with * those where I believe such commitment is wrong
- DateTime: `arrivalTime availabilityEnds availabilityStarts availableFrom availableThrough bookingTime checkinTime checkoutTime contentReferenceTime* coverageEndTime coverageStartTime datasetTimeInterval** dateDeleted* dateIssued* dateRead dateReceived dateSent departureTime ...`
- Date: `birthDate? datePosted* datePublished dateVehicleFirstRegistered deathDate? dissolutionDate expires lastReviewed* ...`
  The birth/death time may be important for modern people.
- Either: `commentTime dateCreated dateModified ...`
  You see here that the name (`time` vs `date`) is picked inconsistently.

**Comment by akuckartz** [comment]
:+1:

**Comment by akuckartz** [comment]
Please consider type dcterms:W3CDTF

**Comment by VladimirAlexiev** [comment]
I think the appropriate xsd type should be used, in order to enable proper/fast value comparison:
- `xsd:dateTime` (mandated by SPARQL)
- `xsd:date` (also supported by most repositories)
- `xsd:gYear, xsd:gYearMonth` (as the case may be)
The last two also show that props should not commit to a certain precision. Eg if you know the birthDate of a painter to a year, it's better to represent it as `"1540"^^xsd:gYear` rather than fake-complete it to `"1540-01-01"^^xsd:date`, which misrepresents the historic facts

**Comment by RichardWallis** [comment]
I agree that there is some confusion in the identifying of expected types for Date and DateTime properties.  Symptomatic perhaps of a community based process that takes input from many sources.

As both [Date](http://schema.org/Date) & [DateTime](http://schema.org/DateTime) are described as expecting values conforming to [ISO 8601](https://en.wikipedia.org/wiki/ISO_8601) the difference between them is somewhat artificial and therefore confusing. 

ISO 8601 is capable of describing things from an individual year or time, through to a fully qualified with timezone datetime to the second, in a way that is compatible with all three accepted serialisation for Schema (Microdata, RDFa, JSON-LD). I believe there is potential for rationalising the datatypes of properties across the vocabulary, whilst maintaining the current flexible yet standards based representation of their values.

**Comment by VladimirAlexiev** [comment]
I'm all **for** flexible representation of values (ideally, marked with the appropriate XSD type: I think these props or datatypes can make such recommendation).
I think that because of that flexibility:
- there shouldn't be props that differ only by representation (precision) of values
- prop names should strive for consistency ("date" or "time"), while not prescribing a certain representation

**Comment by RichardWallis** [comment]
@VladimirAlexiev 
I suppose marking the description of such properties with an equivalent XSD type may help with human comprehension, but the guidance to use a [ISO 8601](https://en.wikipedia.org/wiki/ISO_8601) formatted string is as near enforcement we can get.

- By using a [ISO 8601](https://en.wikipedia.org/wiki/ISO_8601) formatted string both Date & DateTime have the same precision capability.  Confusing I agree.
- I also agree, and ideally we should embark on name rationalisation process.  In practice however with some of the properties in question already in use on 100,000's of sites, I believe it would be considered as not practical because of the disruption caused.

**Comment by VladimirAlexiev** [comment]
I agree that changing existing props may not be the best decision.
If we can't merge these 2 pairs, can we at least deprecate one prop in favor of the other?
